controller/cmd/service-mirror/config_watcher.go

```diff
@@ -105,63 +107,81 @@ func (rcw *RemoteClusterConfigWatcher) Stop() {
 }
 
 func (rcw *RemoteClusterConfigWatcher) registerRemoteCluster(secret *corev1.Secret) error {
-	config, name, domain, err := parseRemoteClusterSecret(secret)
+	config, err := parseRemoteClusterSecret(secret)
 	if err != nil {
 		return err
 	}
 
-	clientConfig, err := clientcmd.RESTConfigFromKubeConfig(config)
+	clientConfig, err := clientcmd.RESTConfigFromKubeConfig(config.apiConfig)
 	if err != nil {
 		return fmt.Errorf("unable to parse kube config: %s", err)
 	}
 
 	rcw.Lock()
 	defer rcw.Unlock()
 
-	if _, ok := rcw.clusterWatchers[name]; ok {
-		return fmt.Errorf("there is already a cluster with name %s being watcher. Please delete its config before attempting to register a new one", name)
+	if _, ok := rcw.clusterWatchers[config.clusterName]; ok {
+		return fmt.Errorf("there is already a cluster with name %s being watcher. Please delete its config before attempting to register a new one", config.clusterName)
 	}
 
-	watcher, err := NewRemoteClusterServiceWatcher(rcw.k8sAPI, clientConfig, name, rcw.requeueLimit, domain)
+	watcher, err := NewRemoteClusterServiceWatcher(rcw.k8sAPI, clientConfig, config.clusterName, rcw.requeueLimit, config.clusterDomain, config.probePort, config.probePath, config.probePeriodSeconds, rcw.enqueueProbeEvent)
 	if err != nil {
 		return err
 	}
 
-	rcw.clusterWatchers[name] = watcher
+	rcw.clusterWatchers[config.clusterName] = watcher
 	watcher.Start()
 	return nil
 
 }
 
 func (rcw *RemoteClusterConfigWatcher) unregisterRemoteCluster(secret *corev1.Secret, cleanState bool) error {
-	_, name, _, err := parseRemoteClusterSecret(secret)
+	config, err := parseRemoteClusterSecret(secret)
 	if err != nil {
 		return err
 	}
 	rcw.Lock()
 	defer rcw.Unlock()
-	if watcher, ok := rcw.clusterWatchers[name]; ok {
+	if watcher, ok := rcw.clusterWatchers[config.clusterName]; ok {
 		watcher.Stop(cleanState)
 	} else {
-		return fmt.Errorf("cannot find watcher for cluser: %s", name)
+		return fmt.Errorf("cannot find watcher for cluser: %s", config.clusterName)
 	}
-	delete(rcw.clusterWatchers, name)
+	delete(rcw.clusterWatchers, config.clusterName)
 
 	return nil
 }
 
-func parseRemoteClusterSecret(secret *corev1.Secret) ([]byte, string, string, error) {
+type watchedClusterConfig struct {
+	apiConfig          []byte
+	clusterName        string
+	clusterDomain      string
+	probePort          int32
```

Comment:
I don't think these probe settings should be part of this cluster config anymore